d3a_api_client/e2e_test/device_aggregator_test.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from d3a_interface.utils import key_in_dict_and_not_none_and_greater_than_zero
+from time import sleep
+
+from d3a_api_client.aggregator import Aggregator
+from d3a_api_client.rest_device import RestDeviceClient
+from d3a_api_client.utils import (
+    get_area_uuid_from_area_name_and_collaboration_id, get_sim_id_and_domain_names)
+
+class TestAggregator(Aggregator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        """
+        Places a bid or an offer whenever a new market is created. The amount of energy
+        for the bid/offer depends on the available energy of the PV, or on the required
+        energy of the load.
+        :param market_info: Incoming message containing the newly-created market info
+        :return: None
```

Comment:
Please add the storage here too or skip the docstring completely. thanks

Reply:
Done